app/models/dialpeer.rb

```diff
@@ -230,6 +237,13 @@ def vendor_owners_the_gateway_group
 
   private
 
+  def prevent_destroy_if_have_pricelist_item
+    if rate_management_pricelist_items.exists?
+      errors.add(:base, 'Has related Rate Management Pricelist Item')
+      throw(:abort)
+    end
+  end
```

Comment:
1. move this to the second commit

in the second commit:
1. prevent destroy only if pricelist_items are not applied
2. nullify dialpeer_id for pricelist_items that are applied